src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/TypeNameCheck.java

```diff
@@ -63,29 +63,54 @@
  * An example of how to configure the check is:
  * </p>
  * <pre>
- * &lt;module name="TypeName"/&gt;
+ * &lt;module name=&quot;TypeName&quot;/&gt;
+ * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface FirstName {} // OK
+ * protected class SecondName {} // OK
+ * enum Third_Name {} // violation, name 'Third_Name' must match pattern '^[A-Z][a-zA-Z0-9]*$'
+ * private class FourthName_ {} // violation, name 'FourthName_'
+ *                              // must match pattern '^[A-Z][a-zA-Z0-9]*$'
  * </pre>
  * <p>
  * An example of how to configure the check for names that begin with
- * a lower case letter, followed by letters, digits, and underscores is:
+ * a lower case letter, followed by letters, digits, and underscores.
+ * Also, suppress the check from being applied to protected and private type:
  * </p>
  * <pre>
- * &lt;module name="TypeName"&gt;
- *   &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
+ * &lt;module name=&quot;TypeName&quot;&gt;
+ *   &lt;property name=&quot;format&quot; value=&quot;^[a-z](_?[a-zA-Z0-9]+)*$&quot;/&gt;
+ *   &lt;property name=&quot;applyToProtected&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;applyToPrivate&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * public interface firstName {} // OK
+ * public class SecondName {} // violation, name 'Second_Name'
```

Comment:
SecondName vs Second_Name

Reply:
lets remove copy paste ... it is already error prone .....
make it like `violation, name must match ....`

Reply:
Done